extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -483,40 +517,57 @@ class LifecycleTask extends BackbeatTask {
                     sortedList.push(deleteMarkers[dmIdx++]);
                 }
             } else {
-                //    NOTE: VersionId may be null
-                const nullVersion = (versions[vIdx].VersionId === 'null'
-                    || deleteMarkers[dmIdx].VersionId === 'null');
-                const isVersionVidNewer = (versions[vIdx].VersionId
-                    < deleteMarkers[dmIdx].VersionId);
-
-                // if there is a null version, handle accordingly
-                if (nullVersion) {
-                    const isVersionLastModifiedNewer = (new Date(versions[vIdx].LastModified)
-                         > new Date(deleteMarkers[dmIdx].LastModified));
-                    const isDMLastModifiedNewer = (new Date(deleteMarkers[dmIdx].LastModified)
-                         > new Date(versions[vIdx].LastModified));
-                    // 2. by LastModified, find newer
-                    if (isVersionLastModifiedNewer) {
-                        sortedList.push(versions[vIdx++]);
-                    } else if (isDMLastModifiedNewer) {
-                        sortedList.push(deleteMarkers[dmIdx++]);
-                    } else {
-                        // 3. choose one randomly since all conditions match
-                        // TODO: to be fixed
-                        sortedList.push(versions[vIdx++]);
-                    }
+                // The `Key` names of the versions and delete markers are the same.
+                // Compare the `LastModified` timestamps of the versions and delete markers.
+                const versionLastModified = new Date(versions[vIdx].LastModified);
+                const deleteMarkerLastModified = new Date(deleteMarkers[dmIdx].LastModified);
+                const isVersionLastModifiedNewer = versionLastModified > deleteMarkerLastModified;
+                const isDMLastModifiedNewer = deleteMarkerLastModified > versionLastModified;
```

Comment:
nit: this makes the code harder to read IMO, as it "hides" the real intent here... i.e. just comparing the date, and handling the 3 possible outcomes of the comparison....